### PR TITLE
issue: 1653789 Fix Rx/Tx migration thread safety

### PR DESCRIPTION
**Pull request by liranoz12**
Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

File: src/vma/proto/dst_entry.h [hunk 1]
```diff
@@ -120,6 +120,7 @@ class dst_entry : public cache_observer, public tostr, public neigh_observer
 	in_addr_t		m_route_src_ip; // source IP used to register in route manager
 	in_addr_t		m_pkt_src_ip; // source IP address copied into IP header
 	lock_mutex_recursive 	m_slow_path_lock;
+	lock_mutex_recursive 	m_tx_migration_lock;
```
Comment (igor-ivanov) [inline src/vma/proto/dst_entry.h]: Do you see any reason to use recursive lock? Please consider using usual lock.

File: src/vma/sock/sockinfo.h [hunk 2]
```diff
@@ -206,6 +206,7 @@ class sockinfo : public socket_fd_api, public pkt_rcvr_sink, public pkt_sndr_sou
 
 	lock_spin_recursive	m_lock_rcv;
 	lock_mutex		m_lock_snd;
+	lock_mutex_recursive	m_rx_migration_lock;
```
Comment (igor-ivanov) [inline src/vma/sock/sockinfo.h]: Do you see any reason to use recursive lock? Please consider using usual lock.

Comment (liranoz12) [reply src/vma/sock/sockinfo.h]: Updated

File: src/vma/proto/dst_entry.h (hunk 1 above)
Comment (liranoz12) [reply src/vma/proto/dst_entry.h]: Updated

**Comment by liranoz12** [comment]
bot:restart

**Comment by liranoz12** [comment]
bot:retest

**Pull request merged**